src/backend/app/tasks/task_routes.py

```diff
@@ -19,6 +19,27 @@
 )
 
 
+@router.get("/", response_model=list[task_schemas.UserTasksStatsOut])
+async def list_tasks(
+    db: Database = Depends(database.get_db),
+    user_data: AuthUser = Depends(login_required),
+):
+    """Get all tasks for a drone user."""
+
+    user_id = user_data.id
+    query = """SELECT role FROM user_profile WHERE user_id = :user_id"""
+    record = await db.fetch_one(query, {"user_id": user_id})
+    if not record:
+        raise HTTPException(status_code=404, detail="User profile not found")
+
+    if record.role != UserRole.DRONE_PILOT.name:
```

Comment:
I don't think we need this. Even Project creators should be allowed to map the tasks.

Reply:
@nrjadkry Thanks for the clarification. I’ve updated the code accordingly.